regression/contracts/assigns_enforce_conditional_function_call_condition/main.c

```diff
@@ -0,0 +1,32 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+bool cond()
+{
+  return true;
+}
+
+int foo(int a, int *x, int *y) __CPROVER_assigns(a && cond() ? *x)
```

Comment:
I am not entirely convinced this is the best syntax to use: it seems that `?` now has a very difference precedence?

Reply:
Frama-C has gone that way, with different precedence rules depending on the context of an expression. I would much like to avoid anything the like.

Reply:
Do you have a suggestion for the operator syntax ? `==>` was considered at some point but there would be similar issues in case the condition contains an even lowest precedence operator like `<==>`.
 Maybe a completely new operator or keyword?

Reply:
What about `:` (colon)?

Reply:
Yes that could work !

How about this pattern:

```C
cond : target
cond: target , ... , target
```
a traditional list of unconditional targets is a special case of 
```C
true: t1, t2, ... 
```
and then separate groups of targets with different conditions using semicolons like so:

```C
__CPROVER_assigns(
t1, t2;
c1: t2, t3;
c2: t4, t5; <-- this one is optional
)
```

Reply:
I'm happy with that if you can come up with a grammar that doesn't require more lookahead!